app/utils/injectReducer.js

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import hoistNonReactStatics from 'hoist-non-react-statics';
+
+import getInjectors from './reducerInjectors';
+
+/**
+ * Dynamically injects a reducer
+ *
+ * @param {string} name Name of the reducer
+ * @param {function} reducer A reducer that will be injected
+ * @param {string} [mode] By default ('restart-on-remount') the reducer will be started on component mount and removed
+ * on component un-mount for improved performance. Another option is 'daemon' that starts the reducer on component mount and
+ * never removes it.
+ */
+export default ({ name, reducer, mode }) => (WrappedComponent) => {
+  class ReducerInjector extends React.Component {
+    static WrappedComponent = WrappedComponent;
+    static contextTypes = {
+      store: PropTypes.object.isRequired,
+    };
+    static displayName = `withReducer(${(WrappedComponent.displayName || WrappedComponent.name || 'Component')})`;
+
+    componentWillMount() {
+      const { injectReducer } = this.injectors;
+
+      injectReducer(name, reducer);
+    }
+
+    componentWillUnmount() {
+      const { ejectReducer } = this.injectors;
```

Comment:
changed the default mode to be 'daemon' (they are not ejected in this mode) and added a warning for `RESTART_ON_REMOUNT`: 

>- `RESTART_ON_REMOUNT`—starts a reducer when a component is being mounted and 
>removes it in `componentWillUnmount` for improved performance. This mode could 
>come in handy for a complex app with a big number of reducers, just be careful
>not to put here sub reducers that are meant to handle actions for the rest of the 
>app's life as they will only function for as long as the component remains on the page.